src/symbolize/symbolizer.rs

```diff
@@ -575,6 +596,26 @@ impl Symbolizer {
     #[cfg_attr(feature = "tracing", crate::log::instrument(skip_all, fields(src = ?src, input = format_args!("{input:#x?}"))))]
     pub fn symbolize_single(&self, src: &Source, input: Input<u64>) -> Result<Symbolized> {
         match src {
+            Source::Apk(Apk {
```

Comment:
just general observation, seems like this symbolize_single vs symbolize (many) introduces a lot of duplication at least with all the validation stuff. Do you think it's possible to abstract that into a single function that would be generic over Input<u64> vs Input<[u64]> ?

Reply:
or maybe symbolize_single() should just call symbolize() directly transforming Input<T> into Input<[T]> ?

Reply:
Fair enough. Will think some more how to decrease code duplication between the two.